src/infer_request.cc

```diff
@@ -1326,6 +1326,7 @@ InferenceRequest::ValidateBytesInputs(
       }
 
       // Start the next element and reset the remaining element size.
+      // FIXME: buffer pointer can be invalid and cause segmentation fault.
```

Comment:
I initially thought the input data were copied to new buffers in `TRITONSERVER_InferenceRequestAppendInputData` but turned out each buffer only holds a reference to memory address. 

The example below will raise segmentation fault because the address of input_data changes when vector resizes. I think it can be a good practice to catch the segfault but not sure how.
```
std::vector<char> input_data;

// push_back some data to input_data
TRITONSERVER_InferenceRequestAppendInputData(
          irequest, "INPUT0", input_data.data(), first_portion_size,
          TRITONSERVER_MEMORY_CPU, 0);

// push_back more data to input_data
TRITONSERVER_InferenceRequestAppendInputData(
          irequest, "INPUT0", input_data.data()+first_portion_size, second_portion_size,
          TRITONSERVER_MEMORY_CPU, 0);

TRITONSERVER_ServerInferAsync(server_, irequest_, nullptr /* trace */);
```

Reply:
I don't think there is programmatic way to detect whether the data pointer passed is valid. This limitation is documented in the [API doc](https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonserver.h#L1289-L1293) so I think that is fine.